pkg/git/repository.go

```diff
@@ -136,10 +138,20 @@ func (o *CloneOptions) Clone(ctx context.Context) (Repository, fs.FS, error) {
 
 	r, err := clone(ctx, o)
 	if err != nil {
-		if err == transport.ErrEmptyRemoteRepository {
-			log.G(ctx).Debug("empty repository, initializing new one with specified remote")
-			r, err = initRepo(ctx, o)
+		if err != transport.ErrRepositoryNotFound && err != transport.ErrEmptyRemoteRepository {
```

Comment:
I would write it with the opposite condition, by searching for specific errors, you can even use a `switch` at that point

Reply:
if there is "repo not found" we need to
1. create repo
2. init local repo

if there is "repo is empty" we need to
1. init local repo

so I wanted to call the init func only once. Maybe it's a bit less clear form the code.